packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -180,7 +190,7 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
   @emit emits "connect" event
   @throws WalletConnectionError
   */
-  async connect(walletName: WalletName): Promise<void> {
+  async connect(walletName: WalletName): Promise<WalletConnectResult> {
```

Comment:
lets not change it, this would actually introduce a breaking change for both dapps and wallet plugins.

Let's actually keep this PR simple and to unblock simply just `return`, we will then design it better

Reply:
why is this a breaking change? I assume existing wallet adapter consumers don't do anything with the connect call result at the moment because it always returns void.

Reply:
changing the return type creates a breaking change

Reply:
ok, reverted to the previous PR state.

Reply:
I'm not sure how this is a breaking change still, what Christian says makes sense to me?

Reply:
it is an optional breaking change from the [wallet plugin side](https://github.com/aptos-labs/wallet-adapter-plugin-template/blob/main/src/index.ts#L26), that can impact dapps. looks like the [interface](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-core/src/types.ts#L56) for this function returns `any`, so it might be ok.